app/src/main/java/com/dgaf/happyhour/Adapter/RestaurantDealAdapter.java

```diff
@@ -0,0 +1,213 @@
+package com.dgaf.happyhour.Adapter;
+
+import android.app.Activity;
+import android.util.Log;
+import android.widget.CompoundButton;
+import android.widget.ImageView;
+import android.widget.TextView;
+import android.widget.Toast;
+import android.widget.ToggleButton;
+
+import com.dgaf.happyhour.Device;
+import com.dgaf.happyhour.Model.AvailabilityModel;
+import com.dgaf.happyhour.Model.DealIcon;
+import com.dgaf.happyhour.Model.DealModel;
+import com.dgaf.happyhour.Model.ModelUpdater;
+import com.dgaf.happyhour.Model.Queries.QueryParameters;
+import com.dgaf.happyhour.R;
+import com.parse.FunctionCallback;
+import com.parse.ParseCloud;
+import com.parse.ParseException;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Created by Adam on 8/26/2015.
+ */
+public class RestaurantDealAdapter implements ModelUpdater<DealModel> {
+    private Activity activity;
+    private DealModel dealModel;
+    private QueryParameters queryParams;
+    private RestaurantDealViewHolder dealHolder;
+
+    public static class RestaurantDealViewHolder implements CompoundButton.OnCheckedChangeListener {
+        public TextView dealTitle;
```

Comment:
RestaurantDealViewHolder is defined within the context of the Adapter. It's also a static class so it's incapable of possesing private members. I've noticed these view holders are common design paradigm in Android but are more relevant for view recycling. They are he closest thing you grt to separating hour view out. That being said it might make more sense to separate them out of the Adapters.